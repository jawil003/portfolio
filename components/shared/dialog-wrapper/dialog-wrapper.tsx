import { css } from "@emotion/react";
import { UniversalPortal } from "@jesstelford/react-portal-universal";
import React from "react";

export interface Props {
  show: boolean;
}

/**
 * An DialogWrapper React Component.
 * @author Jannik Will
 * @version 0.1
 */
export const DialogWrapper: React.FC<Props> = ({
  show,
  children,
}) => {
  return (
    <UniversalPortal selector="#modal">
      <div>
        <div
          css={css`
            & {
              position: fixed;
              top: 0;
              left: 0;
              right: 0;
              bottom: 0;
              min-width: 100vw;
              max-width: 100vw;
              width: 100vw;
              height: 100vh;
              min-height: 100vh;
              max-height: 100vh;
              margin: 0px;
              padding: 0px;
              box-sizing: border-box;
              justify-content: center;
              align-items: center;
              display: ${show
                ? "flex"
                : "none"};
              flex-direction: column;
            }
            &:after {
              content: "";
              position: absolute;
              top: 0;
              left: 0;
              min-width: 100vw;
              max-width: 100vw;
              width: 100vw;
              height: 100vh;
              min-height: 100vh;
              max-height: 100vh;
              background-color: #000000;
              opacity: 0.2;

              z-index: 1;
            }
          `}
        >
          <div
            css={css`
              & {
                box-sizing: border-box;
                background-color: var(
                  --greenblue
                );
                min-width: 400px;
                min-height: 200px;
                border-radius: 10px;
                z-index: 999;
                flex-direction: column;
                padding: 20px;
                margin: -10px 0px;
              }

              & > * {
                margin: 10px 0px;
                color: var(--white);
              }
            `}
          >
            {children}
          </div>
        </div>
      </div>
    </UniversalPortal>
  );
};
