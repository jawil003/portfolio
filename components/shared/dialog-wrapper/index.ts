export { DialogWrapper } from "./dialog-wrapper";
export type { Props as DialogWrapperProps } from "./dialog-wrapper";
