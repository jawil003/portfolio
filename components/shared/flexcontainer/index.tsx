export { FlexContainer } from "./flexcontainer";
export type { Props as FlexContainerProps } from "./flexcontainer";
