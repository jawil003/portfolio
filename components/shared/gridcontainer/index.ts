export { GridContainer } from "./gridcontainer";
export type { Props as GridContainerProps } from "./gridcontainer";
