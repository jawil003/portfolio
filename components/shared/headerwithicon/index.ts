export { HeaderWithIcon } from "./headerwithicon";
export type { Props as HeaderWithIconProps } from "./headerwithicon";
