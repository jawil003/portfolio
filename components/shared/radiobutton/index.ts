export { Radiobutton } from "./radiobutton";
export type { Props as RadiobuttonProps } from "./radiobutton";
