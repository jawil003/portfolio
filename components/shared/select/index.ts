export { Select } from "./select";
export type { Props as SelectProps } from "./select";
