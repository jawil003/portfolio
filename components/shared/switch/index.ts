export { Switch } from "./switch";
export type { Props as SwitchProps } from "./switch";
