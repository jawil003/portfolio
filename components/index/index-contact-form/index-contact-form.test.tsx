import {
  fireEvent,
  render,
  RenderResult,
  waitFor,
} from "@testing-library/react";
import React from "react";
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import * as rIO from "react-intersection-observer";
import { IndexContactForm } from ".";
import ContactRequestService from "../../../services/backend/contactRequest.service";
// eslint-disable-next-line @typescript-eslint/no-unused-vars
import * as Wrapper from "../../shared/dialog-wrapper";

jest.mock(
  "react-intersection-observer",
  () => ({
    __esModule: true,
    useInView: () => ({
      ref: { current: {} },
      inView: true,
    }),
  }),
);

jest.mock(
  "../../shared/dialog-wrapper",
  () => ({
    DialogWrapper: () => {
      return <div />;
    },
  }),
);

describe("IndexContactForm", () => {
  let testForm: RenderResult;
  const testFormData = {
    email: "maxine@musterfrau.de",
    name: "Maxine Musterfrau",
    title: "Help needed",
    message: "Lorem ipsum",
  };

  beforeEach(() => {
    testForm = render(
      <IndexContactForm />,
    );
  });

  test("Create IndexContactForm", () => {
    expect(
      testForm.container.firstChild,
    ).toMatchSnapshot();
  });

  /* test("Submit Form", async () => {
    const spy = await jest
      .spyOn(
        ContactRequestService,
        "do",
      )
      .mockImplementation(
        async () => {},
      );

    const email = await testForm.findByTestId(
      "index-contact-form-email-input",
    );

    const name = await testForm.findByTestId(
      "index-contact-form-name-input",
    );

    const title = await testForm.findByTestId(
      "index-contact-form-title-input",
    );

    const message = await testForm.findByTestId(
      "index-contact-form-message-textarea",
    );

    const submitButton = await testForm.findByTestId(
      "index-contact-form-submit-button",
    );

    waitFor(() => {
      fireEvent.change(email, {
        value: testFormData.email,
      });

      fireEvent.change(name, {
        value: testFormData.name,
      });

      fireEvent.change(title, {
        value: testFormData.title,
      });

      fireEvent.change(message, {
        value: testFormData.message,
      });

      fireEvent.click(submitButton);
    });

    expect(spy).toBeCalledWith(
      testFormData,
    );
  }); */
});
