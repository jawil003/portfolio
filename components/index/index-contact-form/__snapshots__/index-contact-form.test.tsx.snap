// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexContactForm Create IndexContactForm 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex-wrap: no-wrap;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  margin: calc(0px / -2) calc(0px / -2);
}

.emotion-0>* {
  margin: calc(0px / 2) calc(0px / 2);
}

.emotion-1 {
  text-align: center;
  font-style: normal;
  margin: unset;
  -webkit-text-decoration: none;
  text-decoration: none;
  letter-spacing: normal;
  word-spacing: normal;
  color: #000;
  word-wrap: unset;
  padding: unset;
  word-break: normal;
}

.emotion-1 {
  font-weight: 400;
  font-size: 2.125em;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-3.emotion-3>* {
  --indexContactFormRowGap: 0.425em;
  margin: var(
                        --indexContactFormRowGap
                      ) 0;
}

.emotion-3.emotion-3>* {
  width: 100%;
  max-width: 400px;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex-wrap: no-wrap;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  margin: calc(0.6333333333333333em / -2) calc(0px / -2);
}

.emotion-4>* {
  margin: calc(0.6333333333333333em / 2) calc(0px / 2);
}

.emotion-5 {
  font-style: normal;
  font-weight: 600;
  font-size: 1em;
}

.emotion-6 {
  border: 0.13333333333333333em solid #000000;
  box-sizing: border-box;
  border-radius: 1.25em;
  padding: 0.75em 0.8em;
  height: 3.0625em;
}

.emotion-6.emotion-6 {
  margin-bottom: 0;
}

.emotion-6.emotion-6::-webkit-input-placeholder {
  color: #000000;
  font-style: normal;
  font-weight: normal;
  font-size: 1em;
  opacity: 0.5;
}

.emotion-6.emotion-6::-moz-placeholder {
  color: #000000;
  font-style: normal;
  font-weight: normal;
  font-size: 1em;
  opacity: 0.5;
}

.emotion-6.emotion-6:-ms-input-placeholder {
  color: #000000;
  font-style: normal;
  font-weight: normal;
  font-size: 1em;
  opacity: 0.5;
}

.emotion-6.emotion-6::placeholder,
.emotion-6.emotion-6::-webkit-input-placeholder,
.emotion-6.emotion-6::-ms-input-placeholder,
.emotion-6.emotion-6::-moz-placeholder {
  color: #000000;
  font-style: normal;
  font-weight: normal;
  font-size: 1em;
  opacity: 0.5;
}

.emotion-6:focus-within {
  outline: none;
}

.emotion-6 {
  position: relative;
}

.emotion-7 {
  position: absolute;
  top: 0;
  right: 0;
  height: 3.0625em;
  width: 3.0625em;
}

.emotion-7>svg {
  fill: #000;
  padding: 10% 0.8em calc(10% + 0.2em) 10%;
  width: 100%;
  height: 100%;
}

.emotion-8 {
  border: none;
  outline: none;
  width: 100%;
  padding-right: 3.0625em;
}

.emotion-8:focus-within {
  outline: none;
}

.emotion-9.emotion-9 {
  color: red;
  margin: 0;
  font-size: 0.75em;
  line-height: 1em;
  height: 1em;
  margin-top: 0.25em;
  margin-left: 1em;
}

.emotion-23 {
  font-style: normal;
  font-weight: 600;
  font-size: 1em;
}

.emotion-24 {
  border: 0.13333333333333333em solid #000000;
  box-sizing: border-box;
  border-radius: 1.25em;
  padding: 0.75em 0.8em;
  resize: none;
  width: 100%;
}

.emotion-24.emotion-24 {
  margin-bottom: 0;
}

.emotion-24.emotion-24::-webkit-input-placeholder {
  color: #000000;
  font-style: normal;
  font-weight: normal;
  font-size: 1em;
  opacity: 0.5;
}

.emotion-24.emotion-24::-moz-placeholder {
  color: #000000;
  font-style: normal;
  font-weight: normal;
  font-size: 1em;
  opacity: 0.5;
}

.emotion-24.emotion-24:-ms-input-placeholder {
  color: #000000;
  font-style: normal;
  font-weight: normal;
  font-size: 1em;
  opacity: 0.5;
}

.emotion-24.emotion-24::placeholder,
.emotion-24.emotion-24::-webkit-input-placeholder,
.emotion-24.emotion-24::-ms-input-placeholder,
.emotion-24.emotion-24::-moz-placeholder {
  color: #000000;
  font-style: normal;
  font-weight: normal;
  font-size: 1em;
  opacity: 0.5;
}

.emotion-24:focus-within {
  outline: none;
}

.emotion-25.emotion-25 {
  color: red;
  margin: 0;
  font-size: 0.75em;
  line-height: 1em;
  height: 1em;
  margin-top: 0.25em;
  margin-left: 1em;
}

.emotion-26 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-flex-wrap: no-wrap;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  margin: calc(0.6333333333333333em / -2) calc(0px / -2);
}

.emotion-26>* {
  margin: calc(0.6333333333333333em / 2) calc(0px / 2);
}

.emotion-26 {
  display: none;
}

.emotion-28 {
  border: 0.13333333333333333em solid #000000;
  box-sizing: border-box;
  border-radius: 1.25em;
  padding: 0.75em 0.8em;
  height: 3.0625em;
}

.emotion-28.emotion-28 {
  margin-bottom: 0;
}

.emotion-28.emotion-28::-webkit-input-placeholder {
  color: #000000;
  font-style: normal;
  font-weight: normal;
  font-size: 1em;
  opacity: 0.5;
}

.emotion-28.emotion-28::-moz-placeholder {
  color: #000000;
  font-style: normal;
  font-weight: normal;
  font-size: 1em;
  opacity: 0.5;
}

.emotion-28.emotion-28:-ms-input-placeholder {
  color: #000000;
  font-style: normal;
  font-weight: normal;
  font-size: 1em;
  opacity: 0.5;
}

.emotion-28.emotion-28::placeholder,
.emotion-28.emotion-28::-webkit-input-placeholder,
.emotion-28.emotion-28::-ms-input-placeholder,
.emotion-28.emotion-28::-moz-placeholder {
  color: #000000;
  font-style: normal;
  font-weight: normal;
  font-size: 1em;
  opacity: 0.5;
}

.emotion-28:focus-within {
  outline: none;
}

.emotion-30 {
  padding: 0.7em 1em;
  background-color: #00739d;
  color: #f3f3f3;
  outline: none;
  border: none;
  border-radius: 3.125em;
}

.emotion-30:disabled {
  opacity: 0.35;
}

.emotion-30:hover:not(:disabled) {
  background-color: #015b7c;
}

.emotion-30.emotion-30 {
  width: 50%;
}

<div
  class="emotion-0"
  style="opacity: 0;"
>
  <h2
    class="emotion-1 emotion-2"
    color="#000"
    id="heading4"
  >
    Kontaktiere mich gerne
  </h2>
  <div
    style="height: 35px;"
  />
  <form
    action="#"
    class="emotion-3"
    novalidate=""
  >
    <div
      class="emotion-4"
    >
      <label
        class="emotion-5"
        for="email"
      >
        Email Adresse
        *
      </label>
      <div
        class="emotion-6"
      >
        <div
          class="emotion-7"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M22 6c0-1.1-.9-2-2-2H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6zm-2 0l-8 5-8-5h16zm0 12H4V8l8 5 8-5v10z"
            />
          </svg>
        </div>
        <input
          class="emotion-8"
          data-testid="index-contact-form-email-input"
          name="email"
          placeholder="maxine@musterfrau.de"
          required=""
          type="email"
          value=""
        />
      </div>
      <div
        class="emotion-9"
      />
    </div>
    <div
      class="emotion-4"
    >
      <label
        class="emotion-5"
        for="name"
      >
        Name
        *
      </label>
      <div
        class="emotion-6"
      >
        <div
          class="emotion-7"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
        <input
          class="emotion-8"
          data-testid="index-contact-form-name-input"
          name="name"
          placeholder="Maxine Musterfrau"
          required=""
          value=""
        />
      </div>
      <div
        class="emotion-9"
      />
    </div>
    <div
      class="emotion-4"
    >
      <label
        class="emotion-5"
        for="title"
      >
        Titel
        *
      </label>
      <div
        class="emotion-6"
      >
        <div
          class="emotion-7"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 4v3h5.5v12h3V7H19V4z"
            />
          </svg>
        </div>
        <input
          class="emotion-8"
          data-testid="index-contact-form-title-input"
          name="title"
          placeholder="Bitte Titel eingeben"
          required=""
          value=""
        />
      </div>
      <div
        class="emotion-9"
      />
    </div>
    <div
      class="emotion-4"
    >
      <label
        class="emotion-23"
        for="message"
      >
        Nachricht
        *
      </label>
      <textarea
        class="emotion-24"
        data-testid="index-contact-form-message-textarea"
        name="message"
        placeholder="Bitte Nachricht eingeben"
        required=""
        rows="6"
      />
      <div
        class="emotion-25"
      />
    </div>
    <div
      class="emotion-26"
    >
      <label
        class="emotion-5"
        for="hidden"
      >
        Hidden
      </label>
      <input
        class="emotion-28"
        name="hidden"
        value=""
      />
      <div
        class="emotion-9"
      />
    </div>
    <div />
    <button
      class="emotion-30"
      data-testid="index-contact-form-submit-button"
      disabled=""
      type="submit"
    >
      <span>
        Absenden
      </span>
    </button>
  </form>
</div>
`;
